daemon/state.go

```diff
@@ -221,12 +221,13 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState) (resto
 			ctmap.DeleteIfUpgradeNeeded(ep)
 		}
 
+		ep.SetProxy(d.l7Proxy)
```

Comment:
This should be here https://github.com/cilium/cilium/blob/b8aabb4de02e8f2c9e73cf633aa887e73ecbf478/daemon/state.go#L163-L168

Reply:
@aanm the proxy is initiated after the call to `restoreOldEndpoints`:

restoreOldEndpoints is called here:

https://github.com/cilium/cilium/blob/8ff912d66aa8ec7d61cd6e7113ae29411963675e/daemon/daemon.go#L443-L446

proxy is initiated later:

https://github.com/cilium/cilium/blob/8ff912d66aa8ec7d61cd6e7113ae29411963675e/daemon/daemon.go#L517-L525

I could perhaps have a separate way to inject the proxy into the endpoints after L525?

Reply:
Alternatively, I can try starting the proxy before restoring endpoints. The restored endpoints aren't even exposed to the endpointmanager yet when we call `StartProxySupport`, so this should be OK.

Reply:
Shuffled the ordering of initialization of the proxy to be before endpoints are restored.

Reply:
Done!